packages/web3torrent/src/pages/file/File.tsx

```diff
@@ -31,21 +31,27 @@ interface Props {
 
 const File: React.FC<Props> = props => {
   const web3Torrent = useContext(Web3TorrentContext);
-  const [torrent, setTorrent] = useState(parseMagnetURL(useLocation().hash));
+  const {infoHash} = useParams();
+  const queryParams = useQuery();
+  const [torrent, setTorrent] = useState<Torrent>(EmptyTorrent);
   const [, setPeers] = useState({});
   const [loading, setLoading] = useState(false);
   const [buttonLabel, setButtonLabel] = useState('Start Download');
   const [errorLabel, setErrorLabel] = useState('');
 
   const getLiveData = getTorrentAndPeersData(web3Torrent, setTorrent, setPeers);
 
+  useEffect(() => {
+    setTorrent(parseURL(infoHash, queryParams));
+    // eslint-disable-next-line
```

Comment:
Are you sure this is needed? I am not seeing any eslint errors or warnings when I run ` yarn eslint:check` in `web3torrent`

Reply:
In my other PR I [had to add it](https://github.com/statechannels/monorepo/pull/1428/commits/ea8bae6091fd8e8c67c29ef726b4568063a418c2) or it [wouldn't pass the CI tests](https://circleci.com/gh/statechannels/monorepo/18289)

Reply:
resolved with https://github.com/statechannels/monorepo/pull/1427/commits/edc85738a5512fedcc5f8a36419665afe95884d7 and https://github.com/statechannels/monorepo/pull/1427/commits/7e198de4d82917d8a8e2403ec0b1532e6876abcd

Reply:
Thanks! I _think_ you also need to add the following rules for eslint to flag issues correctly:
```
'react-hooks/rules-of-hooks': 'error'
'react-hooks/exhaustive-deps': 'warn'
```

You can add the rules [here](https://github.com/statechannels/monorepo/blob/sp/w3t/magnet/packages/web3torrent/.eslintrc.js#L54).

Reply:
Also, I think you might need to add `eslint-plugin-react-hooks` to dev dependencies and rerun `yarn`.

Reply:
done! :smile: 
Take a look at https://github.com/statechannels/monorepo/pull/1427/commits/9648f9ab425e3bb310ed0ec4b78d4422ae50b678  and https://github.com/statechannels/monorepo/pull/1427/commits/f02939144a5a1d73572bb9a066aab54fcd2a7f39

Reply:
My comments are:
-  the new rules do not belong in the `otherViolations` list as in this case we are turning on eslint rules as opposed to turning off rules that our codebase violates.
- we have adopted the heuristic that package versions are exact. So remove `^` in the package version.